instrumentation/rack/lib/opentelemetry/instrumentation/rack/middlewares/event_handler.rb

```diff
@@ -56,7 +56,9 @@ def on_start(request, _)
 
             parent_context = extract_remote_context(request)
             span = create_span(parent_context, request)
-            request.env[TOKENS_KEY] = register_current_span(span)
+            span_ctx = OpenTelemetry::Trace.context_with_span(span, parent_context: parent_context)
+            rack_ctx = OpenTelemetry::Instrumentation::Rack.context_with_span(span, parent_context: span_ctx)
+            request.env[RACK_OTEL_CONTEXT_KEY] = [OpenTelemetry::Context.attach(rack_ctx), rack_ctx]
```

Comment:
Why do we need both the token and the context here? 👀 Oh, because you need to fetch the span from the context to finish it, and use the token to detach. Can we either document the content of `request.env[RACK_OTEL_CONTEXT_KEY]` or rename the key to something that better describes the content?

Reply:
Naive question: do we need the context? Could we just use the span? I.e.
```ruby
            rack_ctx = OpenTelemetry::Instrumentation::Rack.context_with_span(span, parent_context: parent_context)
            token = OpenTelemetry::Context.attach(rack_ctx)
            request.env[OTEL_TOKEN_AND_SPAN] = [token, span]
```

Reply:
We can just use the span https://github.com/open-telemetry/opentelemetry-ruby-contrib/pull/965/commits/dfcbfb257e2c59965e2f0ee191c523fe019cf0bc

Reply:
My suggestion for using the context may be speculative generality based on the rest of the OTel API always wanting or requiring a context (usually defaulting to `current`). 

No objections from me. If we need the context in the future we can change this.